pkg/providers/imagefamily/nodeimage.go

```diff
@@ -40,8 +50,30 @@ type NodeImageProvider interface {
 	List(ctx context.Context, nodeClass *v1beta1.AKSNodeClass) ([]NodeImage, error)
 }
 
+type provider struct {
+	subscription string
+	location     string
+
+	imageVersionsClient types.CommunityGalleryImageVersionsAPI
+	nodeImageVersions   types.NodeImageVersionsAPI
+
+	nodeImagesCache *cache.Cache
+	cm              *pretty.ChangeMonitor
+}
+
+func NewProvider(versionsClient types.CommunityGalleryImageVersionsAPI, location, subscription string, nodeImageVersionsClient types.NodeImageVersionsAPI) *provider {
+	return &provider{
+		subscription:        subscription,
+		location:            location,
+		imageVersionsClient: versionsClient,
+		nodeImageVersions:   nodeImageVersionsClient,
+		nodeImagesCache:     cache.New(imageExpirationInterval, imageCacheCleaningInterval),
```

Comment:
Consider passing the cache as an argument (could be an optional argument where you new it up here if it's not provided)

Reply:
Updated. Not optional for now.